query-node/mappings/src/content/commentAndReaction.ts

```diff
@@ -0,0 +1,708 @@
+
+  // load same reaction by member to the comment (if any)
+  const existingReactionByMember = await store.get(CommentReaction, {
+    where: { id: commentReactionEntityId({ memberId, commentId, reactionId }) },
+  })
+
+  // load video reaction count by reaction id
+  const reactionsCountByReactionId = await getOrCreateCommentReactionsCountByReactionId(store, comment, reactionId)
+
+  // remove the reaction if same reaction already exists by the member on the comment
+  if (existingReactionByMember) {
+    --reactionsCountByReactionId.count
+    reactionsCountByReactionId.updatedAt = eventTime
+
+    --comment.reactionsCount
+    comment.updatedAt = eventTime
+
+    // remove reaction
+    await store.remove<CommentReaction>(existingReactionByMember)
+  } else {
+    // new reaction
+    const newReactionByMember = new CommentReaction({
+      id: commentReactionEntityId({ memberId, commentId, reactionId }),
+      createdAt: eventTime,
+      updatedAt: eventTime,
+      comment,
+      reactionId,
+      video,
+      memberId: memberId.toString(),
+      member: new Membership({ id: memberId.toString() }),
+    })
+
+    // add reaction
+    await store.save<CommentReaction>(newReactionByMember)
+
+    ++reactionsCountByReactionId.count
+    reactionsCountByReactionId.updatedAt = eventTime
+
+    ++comment.reactionsCount
+    comment.updatedAt = eventTime
+  }
+
+  // save updated comment
+  await store.save<Comment>(comment)
+
+  // save updated reactionsCount by ReactionId
+  await store.save<CommentReactionsCountByReactionId>(reactionsCountByReactionId)
+
+  // common event processing
+
+  const commentReactedEvent = new CommentReactedEvent({
+    ...genericEventFields(event),
+    comment,
+    reactingMember: new Membership({ id: memberId.toString() }),
+    reactionResult: reactionId,
+  })
+
+  await store.save<CommentReactedEvent>(commentReactedEvent)
+}
+
+export async function processCreateCommentMessage(
+  { store, event }: EventContext & StoreContext,
+  memberId: MemberId,
+  message: ICreateComment
+): Promise<void> {
+  // in case of null `parentCommentId` protobuf would assign it a default value i.e. ''
+  const { videoId, parentCommentId, body } = message
+  const eventTime = new Date(event.blockTimestamp)
+
+  // load video
+  const video = await getVideo(store, videoId.toString(), ['channel', 'channel.bannedMembers'])
+  const { channel } = video
+
+  // ensure member is not banned from channel
+  ensureMemberIsNotBannedFromChannel(channel, memberId.toString(), 'Cannot add comment')
+
+  // ensure video's comment section is enabled
+  ensureCommentSectionIsEnabled(video, 'Cannot add comment')
+
+  // if new comment is replying to some parent comment, 1. validate that comment existence,
+  //  2. set `parentComment` to the parent comment, otherwise set `parentComment` to undefined
+  const parentComment =
+    parentCommentId && parentCommentId !== '' ? await getComment(store, parentCommentId.toString()) : undefined
```

Comment:
Redundant expression
```suggestion
    parentCommentId ? await getComment(store, parentCommentId.toString()) : undefined
```

Reply:
Done.